kgraphql/src/main/kotlin/com/apurebase/kgraphql/schema/builtin/BuiltInScalars.kt

```diff
@@ -161,13 +155,19 @@ object BOOLEAN_COERCION : StringScalarCoercion<Boolean> {
         is StringValueNode -> when {
             valueNode.value.equals("true", true) -> true
             valueNode.value.equals("false", true) -> false
-            else -> throw IllegalArgumentException("${valueNode.value} does not represent valid Boolean value")
+            else -> throw InvalidInputValueException(
+                "${valueNode.value} does not represent valid Boolean value",
+                valueNode
+            )
         }
 
         is NumberValueNode -> when (valueNode.value) {
             0L, -1L -> false
```

Comment:
Looking at the [spec](https://spec.graphql.org/draft/#sec-Boolean), `-1` should probably not be considered valid. But also for another PR.